plugins/inputs/rabbitmq/rabbitmq.go

```diff
@@ -350,17 +350,29 @@ func (r *RabbitMQ) requestEndpoint(u string) ([]byte, error) {
 		return nil, err
 	}
 
-	username := r.Username
-	if username == "" {
-		username = DefaultUsername
+	username, err := r.Username.Get()
+	if err != nil {
+		return nil, err
+	}
+	defer username.Destroy()
+	usernameStr := username.String()
+
+	if usernameStr == "" {
+		usernameStr = DefaultUsername
+	}
```

Comment:
You can directly ask the secret if it is empty...
```suggestion
	username := DefaultUsername
	if !r.Username.Empty() {
		usr, err := r.Username.Get()
		if err != nil {
			return nil, err
		}
		defer usr.Destroy()
		username = usr.String()
	}
```
What do you think? If you like it, the same can be done for password.

Reply:
Looks good, I've updated the PR.